src/components/autozoomControl.ts

```diff
@@ -93,11 +97,15 @@ export default class AutozoomControl implements ICnnControl {
   }
 
   private _validateOptions(options: AutozoomControlOpts) {
-    if (
-      options.shouldAutoFrame === false &&
-      options.mode &&
-      options.mode !== AutozoomModes.NORMAL
-    ) {
+    if (!options.mode || !Object.values(AutozoomModes).includes(options.mode)) {
+      throw Error(`The following mode is not supported on autozoom controller: ${options.mode}`);
+    }
+
+    if (options.shouldAutoFrame === undefined || options.shouldAutoFrame === null) {
+      throw Error(`Option 'shouldAutoFrame' cannot not be set to ${options.shouldAutoFrame}`);
+    }
+
+    if (options.shouldAutoFrame === false && options.mode !== AutozoomModes.NORMAL) {
```

Comment:
`options.shouldAutoFrame === false` .. :)

Reply:
Not sure I understand your suggestion here!

Reply:
Got it, thanks for your review Eskil :D